Sources/Web3ModalUI/Resources/Extensions/Font+extension.swift

```diff
@@ -1,19 +1,118 @@
-public extension Font {
-    static var large500: Font = .custom("SF Pro Text", size: 20.0).weight(.regular)
-    static var large600: Font = .custom("SF Pro Text", size: 20.0).weight(.medium)
-    static var large700: Font = .custom("SF Pro Text", size: 20.0).weight(.semibold)
-    static var micro600: Font = .custom("SF Pro Text", size: 10.0).weight(.bold)
-    static var micro700: Font = .custom("SF Pro Text", size: 10.0).weight(.medium)
-    static var paragraph500: Font = .custom("SF Pro Text", size: 16.0).weight(.regular)
-    static var paragraph600: Font = .custom("SF Pro Text", size: 16.0).weight(.medium)
-    static var paragraph700: Font = .custom("SF Pro Text", size: 16.0).weight(.semibold)
-    static var small500: Font = .custom("SF Pro Text", size: 14.0).weight(.regular)
-    static var small600: Font = .custom("SF Pro Text", size: 14.0).weight(.medium)
-    static var tiny500: Font = .custom("SF Pro Text", size: 12.0).weight(.regular)
-    static var tiny600: Font = .custom("SF Pro Text", size: 12.0).weight(.medium)
-    static var title500: Font = .custom("SF Pro Text", size: 24.0).weight(.regular)
-    static var title600: Font = .custom("SF Pro Text", size: 24.0).weight(.medium)
-    static var title700: Font = .custom("SF Pro Text", size: 24.0).weight(.semibold)
+public enum W3MFont {
+    case large500
+    case large600
+    case large700
+    case micro600
+    case micro700
+    case paragraph500
+    case paragraph600
+    case paragraph700
+    case small500
+    case small600
+    case tiny500
+    case tiny600
+    case title500
+    case title600
+    case title700
+    
+    var weight: Font.Weight {
+        switch self {
+        case .large500: return .regular
+        case .large600: return .medium
+        case .large700: return .semibold
+        case .micro600: return .medium
+        case .micro700: return .semibold
+        case .paragraph500: return .regular
+        case .paragraph600: return .medium
+        case .paragraph700: return .semibold
+        case .small500: return .regular
+        case .small600: return .medium
+        case .tiny500: return .regular
+        case .tiny600: return .medium
+        case .title500: return .regular
+        case .title600: return .medium
+        case .title700: return .semibold
+        }
+    }
+    
+    var size: CGFloat {
+        switch self {
+        case .large500: return 20.0
+        case .large600: return 20.0
+        case .large700: return 20.0
+        case .micro600: return 10.0
+        case .micro700: return 10.0
+        case .paragraph500: return 16.0
+        case .paragraph600: return 16.0
+        case .paragraph700: return 16.0
+        case .small500: return 14.0
+        case .small600: return 14.0
+        case .tiny500: return 12.0
+        case .tiny600: return 12.0
+        case .title500: return 24.0
+        case .title600: return 24.0
+        case .title700: return 24.0
+        }
+    }
+}
+
+struct ScaledFont: ViewModifier {
```

Comment:
it's scale font depends on screen size or how it's working?

Reply:
All the fonts are "Scaled". The problem here is if you want a `.system`  font of a specific size and weight, it no longer reacts to DynamicType size automatically, and you need to do this hack.  

More context here https://www.hackingwithswift.com/quick-start/swiftui/how-to-use-dynamic-type-with-a-custom-font

in our case we don't have a custom but system. If we don't do this we would need to use https://developer.apple.com/documentation/swiftui/font/custom(_:size:relativeto:)